src/swift/SwiftAPIViewCore/Sources/Models/Declarations/ExtensionModel.swift

```diff
@@ -0,0 +1,104 @@
+// --------------------------------------------------------------------------
+//
+// Copyright (c) Microsoft Corporation. All rights reserved.
+//
+// The MIT License (MIT)
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the ""Software""), to
+// deal in the Software without restriction, including without limitation the
+// rights to use, copy, modify, merge, publish, distribute, sublicense, and/or
+// sell copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED *AS IS*, WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+// FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
+// IN THE SOFTWARE.
+//
+// --------------------------------------------------------------------------
+
+import Foundation
+import AST
+
+
+class ExtensionModel: Tokenizable, Commentable {
+
+    var lineId: String?
+    var attributes: AttributesModel
+    var accessLevel: AccessLevelModifier?
+    var typeModel: TypeModel
+    var typeInheritanceClause: TypeInheritanceModel?
+    var genericWhereClause: GenericWhereModel?
+    var members: [AccessLevelProtocol]
+
+    init(from decl: ExtensionDeclaration, parent: Linkable) {
+        lineId = identifier(forName: decl.type.textDescription, withPrefix: parent.definitionId)
+        attributes = AttributesModel(from: decl.attributes)
+        accessLevel = decl.accessLevel
+        self.typeModel = decl.type.toTokenizable()!
+        typeInheritanceClause = TypeInheritanceModel(from: decl.typeInheritanceClause)
+        genericWhereClause = GenericWhereModel(from: decl.genericWhereClause)
+        members = [AccessLevelProtocol]()
+        decl.members.forEach { member in
+            switch member {
+            case let .declaration(decl):
+                // TODO: We need to push these into the relevant object definitions.
```

Comment:
What does this mean? Push the extension declaration into the proper Extended class?

Reply:
It was more me thinking out loud. Part of: https://github.com/Azure/azure-sdk-tools/issues/2536


I'm not yet sure what the best way would be.